### ISSUE TITLE
Create meaningful unit tests

### ISSUE THREAD

**Issue by majkamichal** [el-meyer/airship#143]
It would be maybe beneficial to have a set of meaningful unit tests to make sure some type of changes do not break the code.

`testthat` functionality is built-in into the RStudio interface.

**Comment by el-meyer** [comment]
Yes that would be great. I have not yet figured out how to properly do that for Shiny Apps, e.g. make sure that a different version of reading the data always produces meaningful summarized data or making sure that the Download button works

**Comment by majkamichal** [comment]
Looks to be a great resource: https://rstudio.github.io/shinytest2/index.html 
When I was learning shiny back in 2015 there were not so many such great tools! :)

**Comment by majkamichal** [comment]
There is also a first version that seems to be stable: https://github.com/rstudio/shinytest/

One large language model explains differences in the following way:

------------------------------------------------------

The `shinytest` and `shinytest2` are two separate packages in R for testing Shiny apps. 

The original `shinytest` package provides an end-to-end testing framework for Shiny apps. This includes recording and replaying user input and output, and verifying that the app behaves in the expected way. The package is based on the `Selenium` browser automation tool.

On the other hand, `shinytest2` is an updated version of `shinytest` that provides improved error messages and better integration with other Shiny packages, such as `shinydashboard` and `shinyWidgets`. It also includes improved support for Shiny apps that use HTML widgets and JavaScript code. The package is based on the `puppeteer` testing framework.

In summary, both packages aim to provide Shiny app testing by automating user inputs and checking for expected outputs. One key difference is that `shinytest2` offers more advanced features than `shinytest`, thanks to being based on the `puppeteer` testing framework. However, `shinytest` still has its uses, particularly if you need to use the `Selenium` browser automation tool or if you want a simpler set of features.

Ultimately, your choice of which package to use will depend on the specific needs of your project and the features that each package provides.
